rs/https_outcalls/consensus/src/pool_manager.rs

```diff
@@ -291,9 +291,21 @@ impl CanisterHttpPoolManagerImpl {
             return Vec::new();
         };
 
+        let existing_signed_requests: BTreeSet<_> = canister_http_pool
```

Comment:
I think we would also have to extend this set whenever we decide to move a share to validated. Otherwise we might still validate multiple shares, as long as they are part of the same unvalidated "batch".
```suggestion
        let mut existing_signed_requests: BTreeSet<_> = canister_http_pool
```

Reply:
Good catch Leo, thank you!